firmware/components/driver_framebuffer/driver_framebuffer_matrix.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include "sdkconfig.h"
+#include "include/driver_framebuffer_internal.h"
+#include <math.h>
+
+#ifdef CONFIG_DRIVER_FRAMEBUFFER_ENABLE
+
+#ifdef CONFIG_G_MATRIX_ENABLE
+
+/*
+ * This file is NOT part of an external library but is rather created and maintained by the badge team.
+ * 
+ * DO NOT TOUCH THIS FILE UNLESS YOU UNDERSTAND MATRICES!
+ */
+
+//creates a 2D matrix representing the given rotation in radians
+matrix_2d matrix_2d_rotate(float angle) {
+    float dSin = sin(angle);
+    float dCos = cos(angle);
+    return (matrix_2d){ .arr = {
+            dCos, -dSin, 0,
+            dSin, dCos, 0
+    }};
+}
+
+//creates a 2D matrix representing the given translation
+matrix_2d matrix_2d_translate(float x, float y) {
+    return (matrix_2d){ .arr = {
+            1, 0, x,
+            0, 1, y
+    }};
+}
+
+//creates a 2D matrix representing the given scaling
+matrix_2d matrix_2d_scale(float x, float y) {
+    return (matrix_2d){ .arr = {
+            y, 0, 0,
+            0, x, 0
+    }};
+}
+
+// Potentially.
+// matrix_3x2_2d matrix_2d_shear(float x, float y);
+
+/* MATRIX OPERATIONS */
+
+//checks whether or not the matrix is an identity matrix
+bool matrix_2d_is_identity(matrix_2d matrix) {
+    return matrix.var.a0 == 1 && matrix.var.a1 == 0 && matrix.var.a2 == 0 &&
+           matrix.var.b0 == 0 && matrix.var.b1 == 1 && matrix.var.b2 == 0;
+}
+
+/*
+ * 
+ * Matrix multiplication:
+ * 
+ * [a  b  c] [p  q  r] [ap + bs + cv   aq + bt + cw   ar + bu + cx]
+ * [d  e  f]*[s  t  u]=[dp + es + fv   dq + et + fw   dr + eu + fx]
+ * [g  h  i] [v  w  x] [gp + hs + iv   gq + ht + iw   gr + hu + ix]
+ * 
+ * [a  b  c] [p]  [a]  [b]  [c] [ap + bs + cv]
+ * [d  e  f]*[s]=p[d]+s[e]+v[f]=[dp + es + fv]
+ * [g  h  i] [v]  [g]  [h]  [i] [gp + hs + iv]
+ * 
+ * [a  b  c] [q]  [a]  [b]  [c] [aq + bt + cw]
+ * [d  e  f]*[t]=q[d]+t[e]+w[f]=[dq + et + fw]
+ * [g  h  i] [w]  [g]  [h]  [i] [gq + ht + iw]
+ * 
+ * [a  b  c] [r]  [a]  [b]  [c] [ar + bu + cx]
+ * [d  e  f]*[u]=r[d]+u[e]+x[f]=[dr + eu + fx]
+ * [g  h  i] [x]  [g]  [h]  [i] [gr + hu + ix]
```

Comment:
These are all matrix/vector multiplication, and I'm not sure it's helpful for the understanding of newbies to have them duplicated like this

Reply:
I think i want to keep this, because it's a handy cheatsheet for me :P